src/playback-watcher.js

```diff
@@ -115,6 +115,24 @@ export default class PlaybackWatcher {
       this.tech_.on(['seeked', 'seeking'], loaderChecks[type].reset);
     });
 
+    // handle seeking into a gap
+    this.tech_.on('seeking', () => {
+      let seekTimeout;
+
+      const clear = () => {
+        this.tech_.off('seeking', clear);
+        this.tech_.off('seeked', clear);
+        window.clearTimeout(seekTimeout);
+      };
+
+      this.tech_.on('seeking', clear);
+      this.tech_.on('seeked', clear);
+
+      seekTimeout = window.setTimeout(() => {
```

Comment:
We might want to listen for segment loader appends here, rather than waiting 3s as that might not be enough time on slow connections. I also think that we should be checking if the player is buffering or not, but techWaiting might do that.

Reply:
I think we should go with what you said here about waiting for segment loader appends rather than a timeout. I think a timeout alone might get us into trouble with some of our potential resolution strategies.